wicket-core/src/main/java/org/apache/wicket/Session.java

```diff
@@ -915,6 +924,10 @@ public int nextPageId()
 	 */
 	public final IPageManager getPageManager()
 	{
+		if (Boolean.TRUE.equals(RequestCycle.get().getMetaData(PAGES_UNLOCKED))) {
```

Comment:
When detached, Session no longer gives access to it's access-synchronizing manager.